### PR TITLE
Add issue application about to submit with mini application creation

### PR DESCRIPTION
**Pull request by mapingo**
### Change description ###
Add issue application about to submit callback
Add mini application pdf creation

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/IssueApplicationService.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@Service
+public class IssueApplicationService {
+
+    @Autowired
+    private MiniApplication miniApplication;
+
+    @Autowired
+    private CaseDataUpdaterChainFactory caseDataUpdaterChainFactory;
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseData caseData,
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/IssueApplicationService.java:27]: See previous note about returning AboutToStartOrSubmitResponse

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/updater/MiniApplication.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            DIVORCE_MINI_APPLICATION_DOCUMENT_NAME
+        );
+
+        final var ccdDocument = new Document(
+            documentInfo.getUrl(),
+            documentInfo.getFilename(),
+            documentInfo.getBinaryUrl()
+        );
+
+        final var divorceDocument = DivorceDocument
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/updater/MiniApplication.java:70]: Seems like a few hoops to jump through here converting the return type of `docAssemblyService.renderDocument` to a `DivorceDocument`. Maybe we should make a utility function for that.

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerIssueApplication.java [hunk 3]
```diff
@@ -21,32 +29,57 @@
...
-                .optional(MarriageDetails::getDate)
-                .optional(MarriageDetails::getApplicant1Name)
-                .optional(MarriageDetails::getApplicant2Name)
-                .mandatory(MarriageDetails::getPlaceOfMarriage)
-                .done();
+            .optional(MarriageDetails::getDate)
+            .optional(MarriageDetails::getApplicant1Name)
+            .optional(MarriageDetails::getApplicant2Name)
+            .mandatory(MarriageDetails::getPlaceOfMarriage)
+            .done();
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerIssueApplication.java:66]: formatting ?

File: src/main/java/uk/gov/hmcts/divorce/document/CaseDataDocumentService.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public CaseData renderDocumentAndUpdateCaseData(final CaseData caseData,
+                                                    final DocumentType documentType,
+                                                    final Supplier<Map<String, Object>> templateContentSupplier,
+                                                    final Long caseId,
+                                                    final String authorisation,
+                                                    final String templateId,
+                                                    final String documentName,
+                                                    final LanguagePreference languagePreference) {
+
+        final var documentInfo = docAssemblyService.renderDocument(
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/document/CaseDataDocumentService.java:34]: some logging would be nice like rendering the document for case id

File: src/test/java/uk/gov/hmcts/divorce/caseworker/service/CaseworkerIssueApplicationServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    private MiniApplication miniApplication;
+
+    @Mock
+    private CaseDataUpdaterChainFactory caseDataUpdaterChainFactory;
+
+    @InjectMocks
+    private IssueApplicationService issueApplicationService;
+
+    @Test
+    void shouldCompleteStepsToCreateApplication() {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/caseworker/service/CaseworkerIssueApplicationServiceTest.java:40]: issue application and not create ?

File: src/test/java/uk/gov/hmcts/divorce/caseworker/service/CaseworkerIssueApplicationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.divorce.testutil.TestConstants.TEST_AUTHORIZATION_TOKEN;
+import static uk.gov.hmcts.divorce.testutil.TestConstants.TEST_CASE_ID;
+import static uk.gov.hmcts.divorce.testutil.TestDataHelper.LOCAL_DATE;
+
+@ExtendWith(MockitoExtension.class)
+class CaseworkerIssueApplicationServiceTest {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/caseworker/service/CaseworkerIssueApplicationServiceTest.java:28]: should we include unhappy tests ?

Comment (mapingo) [reply src/test/java/uk/gov/hmcts/divorce/caseworker/service/CaseworkerIssueApplicationServiceTest.java:28]: I would not add any unhappy path tests here as the only exceptions would be Runtime ones, and they should be tested at in the integration test level

Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/divorce/caseworker/service/CaseworkerIssueApplicationServiceTest.java:28]: yeah makes sense

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**